meson.build

```diff
@@ -55,6 +61,21 @@ mconfig_data.set('DINIT_VERSION', '"' + meson.project_version() + '"')
 mconfig_data.set('SYSCONTROLSOCKET', '"' + dinit_control_socket_path + '"')
 mconfig_data.set('SBINDIR', '"' + sbindir + '"')
 mconfig_data.set('SHUTDOWN_PREFIX', '"' + shutdown_prefix + '"')
+mconfig_data.set('DEFAULT_AUTO_RESTART', default_auto_restart)
+if '.' in default_start_timeout
```

Comment:
This is clunky (checking for '.' and handling it differently in either case, plus repeating that for several different things). Can this be simplified (even by dropping support for sub-second times?)

Also, I just realised meson builds aren't useing mconfig-gen to generate mconfig.h. I think that it would be better to use mconfig-gen; it can do a lot more sanity checking a lot easier, and it generates constants where appropriate instead of always `#define`'s.

Reply:
Hmm, I think nobody use sub-second times in default times, so it's reasonable to remove it.

About meson, I like to talk about it later (after resolving some other things).